designprep/src/main/java/SnakeAndLadderGame.java

```diff
@@ -0,0 +1,131 @@
+import gamemodel.GameEnhancement;
+import gamemodel.GameModel;
+import gamemodel.Player;
+
+import java.util.*;
+
+/**
+ * @author r.shukla
+ */
+public class SnakeAndLadderGame {
+
+    private static final int DICE_FACES = 6;
+    private static Map<Integer, GameModel> snakesMap = new HashMap<>();
+    private static Map<Integer, GameModel> ladderMap = new HashMap<>();
+    private static Queue<Player> playerQueue = new LinkedList<>();
+    private static final int MAX_BOARD_CELL = 100;
+
+    /**
+     * ENUM for game factors
+     */
+    private enum GameFactors { SNAKES, LADDERS }
+
+    /**
+     *
+     * @param args
+     */
+    public static void main(String[] args) {
+        scannGameInfo();
+        playGame();
+    }
+
+    private static void scannGameInfo() {
+        Scanner scanner = new Scanner(System.in);
+        int snakes  = scanner.nextInt();
+        while (snakes > 0) {
+            putValueInMap(GameFactors.SNAKES, scanner);
+            snakes--;
+        }
+        int ladders = scanner.nextInt();
+        while (ladders > 0) {
+            putValueInMap(GameFactors.LADDERS, scanner);
+            ladders--;
+        }
+        int players = scanner.nextInt();
+        while (players > 0) {
+            Player player = new Player(scanner.next());
+            playerQueue.add(player);
+            players--;
+        }
+    }
+
+    private static void putValueInMap(GameFactors type, Scanner scanner) {
+        int startPosition = scanner.nextInt();
+        int endPosition = scanner.nextInt();
+        if (GameFactors.SNAKES.equals(type)) {
+            snakesMap.put(startPosition, getGameModel(startPosition, endPosition));
+        } else {
+            ladderMap.put(startPosition, getGameModel(startPosition, endPosition));
+        }
+    }
+
+    private static GameModel getGameModel(int startPosition, int endPosition) {
+        return new GameEnhancement(startPosition, endPosition);
+    }
+
+    //Used single class , can be moved in separate class for game algo
+    /**
+     * made public to test the functionality , can be removed in general as the method should be accessed from main method only
+     */
+    public static void playGame() {
+        while (!playerQueue.isEmpty()) {
+            Player player = playerQueue.poll();
+            int diceValue = rollDice();
+            int playerPosition = setPlayerPosition(player, diceValue);
+            playerQueue.add(player);
+            if (checkHasPlayerWon(playerPosition)) {
```

Comment:
should be hasPlayerWon instead of checkHasPlayerWon since it checks and returns and not just only check.